### DISCUSSION TITLE
v1.2.0b5

### DISCUSSION THREAD

**Discussion by JosephHewitt** [JosephHewitt/wardriver_rev3#140] [Announcements]
### Beta Release version 1.2.0b5

This version is a beta and is not yet considered to be stable.

### Changes since 1.2.0b4:

- Improve WiFi scanning speed: #138 
- Fix BLE count on LCD: #137
- Log/display if Side B crashes: #137
- Improve speed of hashing files: #136
- Add advanced config to auto reset, and override GPS coordinates: #134
- Improvements to WiGLE connection and upload speeds: #133 & #128
- Add buttons to choose desired branch (stable/beta) to OTA update: #132
- Add WiGLE automatic upload functionality: #130
- Fix crash which occurred when accessing some files: #129
- Fix display rotation not working: #126
- Stop prompting to download/install Side B updates multiple times: #125
- Allow logging networks over UART for PC-tethering, even without an SD card: #121

### Important note for users on previous betas:

Many of the previous releases contained issues with OTA updates. This release has no known problems which can prevent OTA updates from being installed. It is recommended to install this update manually if you are experiencing problems updating. It is recommended to update both **A** and **B** together for this release.

### Known issues

- You may be prompted to install an already-installed update for Side B if you installed it **manually** (with Arduino IDE, esptool, etc) or after performing a factory reset. This will stop happening once you install any update for Side B using the OTA function.

<hr /><em>This discussion was created from the release <a href='https://github.com/JosephHewitt/wardriver_rev3/releases/tag/v1.2.0b5'>v1.2.0b5</a>.</em>

**Comment by pejacoby** [comment]
I hit the issues updating Side B based on manual install of beta 4.  The full process ran but then as Side B completed I got "Failure: Hash mismatch | Update failed".  Side A worked fine and is now on b5.

I ran through the manual upload process and installed B.bin from the build.zip and still got the hash failure.

It would be good to add a pointer to the OTA process on the wiki https://wardriver.uk/ota to the Github page so the instructions on how to update via all methods are there for reference.  I'll admit I didn't think of the Wiki immediately

**Reply by JosephHewitt** [reply]
Do you know which version B is currently at? That issue shouldn't affect you if it is already on b4, so I'm curious to find out if there is another issue causing the same problem.

The right process here would be to update B manually (Arduino IDE, esptool, etc) since this issue will likely stop most updates from working. After the manual install, you will likely still get prompts to update B. It is safe to tell it to update as long as it states 1.2.0b5; this will install the same version a second time but will make the update prompt go away. Or you can ignore the update prompt until the next update.

**Reply by pejacoby** [reply]
I'm on beta 4, installed manually from the last build.  

I just did a manual update to b5 on side B, and then ran through the OTA update which showed b5 available as I had /B.bin on the SD card already. I saw the hash failure again, but I know I'm on b5 on side B now, so I deleted the B.bin file until next time.


11:21:39.809 -> GET /fwins?h=2e62ab8ac3525aaa0634f8ecfa40707dea6246b39c5ad910fefdf4f9de45808e&n=/B.bin HTTP/1.1
11:21:39.841 -> Host: 192.168.1.188
11:21:39.841 -> Connection: keep-alive
11:21:39.841 -> Upgrade-Insecure-Requests: 1
11:21:39.841 -> Accept: text/html,application/xhtml+xml,application/xml;q=0.9,*/*;q=0.8
11:21:39.841 -> User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/17.1 Safari/605.1.15
11:21:39.841 -> Referer: http://192.168.1.188/fwup
11:21:39.841 -> Accept-Language: en-US,en;q=0.9
11:21:39.841 -> Accept-Encoding: gzip, deflate
11:21:39.874 -> 
11:21:39.874 -> 
11:21:39.874 -> Firmware install requested: /B.bin2e62ab8ac3525aaa0634f8ecfa40707dea6246b39c5ad910fefdf4f9de45808eWill install firmware
11:21:39.874 -> Validating firmware
11:22:13.984 -> Contacting OTA server -> /hashes/2e62ab8ac3525aaa0634f8ecfa40707dea6246b39c5ad910fefdf4f9de45808e.txt
11:22:13.984 -> primary cert
11:22:15.268 -> HTTP/1.1 200 OK
11:22:15.268 -> Date: Sun, 19 Nov 2023 17:22:15 GMT
11:22:15.268 -> Server: Apache
11:22:15.268 -> Last-Modified: Sun, 19 Nov 2023 17:00:02 GMT
11:22:15.268 -> ETag: "66-60a844cea27da"
11:22:15.268 -> Accept-Ranges: bytes
11:22:15.268 -> Content-Length: 102
11:22:15.268 -> CL:102
11:22:15.268 -> Vary: Accept-Encoding
11:22:15.268 -> Connection: close
11:22:15.300 -> Content-Type: text/plain
11:22:15.300 -> 
11:22:15.300 -> END OF HEADER^
11:22:15.300 -> OTA contact end
11:22:15.300 -> Got OTA hash check response:
11:22:15.300 -> OKHASH>2e62ab8ac3525aaa0634f8ecfa40707dea6246b39c5ad910fefdf4f9de45808e
11:22:15.300 -> VERS>1.2.0b5
11:22:15.300 -> DATE>Nov-19-2023
11:22:15.300 -> 
11:22:15.300 -> Firmware update for side B
11:22:15.300 -> Getting B ready
...
11:22:17.078 -> B is ready
11:25:36.305 -> B confirmed failure
11:25:46.314 -> Update failed

**Reply by JosephHewitt** [reply]
Just to confirm, for the manual install were you using the Arduino IDE or esptool?

**Reply by pejacoby** [reply]
Installing via esptool.py v4.6.1 using the instructions here.

**Reply by JosephHewitt** [reply]
Interesting that you got the hash mismatch error again then. That could indicate communication problems between the 2 ESP32s. It could be another bug but I didn't have this problem when I was testing the last 2 releases so unfortunately this might take some time to track down.

**Reply by pejacoby** [reply]
If it's relevant, my last four startups as seen in test.txt, note the bsh hash from side B is just "xxxxx"

_BOOT_1.2.0b5, ut=2390117, rr=1, id=3893376
_BOOT_1.2.0b5, ut=2391919, rr=1, id=3893376, bc=264, ep=1700415119, bsh=xxxxx
_BOOT_1.2.0b5, ut=2407728, rr=1, id=3893376, bc=265, ep=1700417804, bsh=xxxxx
_BOOT_1.2.0b5, ut=2403219, rr=1, id=3893376

**Reply by JosephHewitt** [reply]
The hash there is only updated after a successful OTA update so it won't change if you update manually.

**Reply by pejacoby** [reply]
I guessed as much, will keep trying as things move along. Managed to install B.ino on side A by accident, so proved that display code still works :)

If I attach the serial console to side B during updates will I see anything different?  I could give that a try.
